aspnetcore/fundamentals/minimal-apis/openapi.md

```diff
@@ -77,52 +77,143 @@ ASP.NET Core provides the [`Microsoft.AspNetCore.OpenApi`](https://www.nuget.org
+    .Produces<IList<Todo>>();
+```
 
-### Add tags to the Open API description
+Leveraging `TypedResults` in the implementation of an endpoint's route handler will automatically include the response type metadata for the endpoint. For example, the code below will automatically annotate the endpoint with a response under the `200` status code with an `application/json` content type.
 
-The following code uses an [OpenAPI grouping tag](https://swagger.io/docs/specification/grouping-operations-with-tags/):
+```csharp
+app.MapGet("/todos", async (TodoDb db) =>
+{
+    var todos = await db.Todos.ToListAsync());
+    return TypedResults.Ok(todos);
+});
+```
 
-[!code-csharp[](samples/todo/Program.cs?name=snippet_grp)]
+#### Set responses for ProblemDetails
 
-### Add endpoint summary or description
+When setting the response type for endpoints that may return a ProblemDetails response, the [`ProducesProblem`](/dotnet/api/microsoft.aspnetcore.http.openapiroutehandlerbuilderextensions.producesproblem) extension method or [`TypeResults.Problem`](/dotnet/api/microsoft.aspnetcore.http.typedresults.problem?) can be used to add the appropriate annotation to the endpoint's metadata.
 
-The following highlighted code provides summary and description text for OpenAPI spec generation. You can call extension methods or use attributes:
+When there are no explicit annotations provided by one of the strategies above, the framework will attempt to determine a default response type by examining the signature of the response. This default response will be populated under the `200` status code in the OpenAPI definition.
 
-:::code language="csharp" source="endpoint-description-summary/7.0-samples/OpenAPISummary/Program.cs" id="summary_and_description" highlight="13-14":::
+### Describe request body and parameters
 
-:::code language="csharp" source="endpoint-description-summary/7.0-samples/OpenAPISummary/Program.cs" id="summary_and_description2" highlight="2-3":::
+In addition to describing the types that are returned by an endpoint, OpenAPI also supports annotating the inputs that are consumed by an API. These inputs fall into two categories:
+
+* Parameters that appear in the path, query string, headers, or cookies
+* Data transmitted as part of the request body
+
+The framework will infer the types for request parameters in the path, query, and header string automatically based on the signature of the route handler.
+
+To define the type of inputs transmitted as the request body, you can configure the properties by using the `Accepts` extension method to define the object type and content type that are expected by the request handler.
+
+```csharp
+app.MapPost("/todos/{id}", (int id, Todo todo) => ...)
+  .Accepts<Todo>();
+```
+
+When there is no explicit annotation provide, the framework will attempt to determine the default request type if there exists a request body parameter in the endpoint handler. 
```

Comment:
This might be clearer with an example, e.g. the previously shown example for `Accepts` isn't actually required given that accepting the `Todo` parameter (assuming it doesn't implement `BindAsync`) will default to inferring accepts metadata of that shape and application/json. Also consider mentioning `IEndpointParameterMetadata` as an additional strategy that's used to acquire accepts metadata.